extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -74,6 +74,37 @@ class AmpInstagram extends AMP.BaseElement {
         this.element.getAttribute('shortcode')),
         'The data-shortcode attribute is required for <amp-instagram> %s',
         this.element);
+
+    this.buildPrerenderDom_();
+  }
+
+  /** @private */
+  buildPrerenderDom_() {
+    const image = this.getWin().document.createElement('amp-img');
+    // This will redirect to the image URL. By experimentation this is
+    // always the same URL that is actually used inside of the embed.
+    image.setAttribute('src', 'https://www.instagram.com/p/' +
+        encodeURIComponent(this.shortcode_) + '/media/?size=l');
+    image.setAttribute('width', this.element.getAttribute('width'));
+    image.setAttribute('height', this.element.getAttribute('height'));
+    image.setAttribute('layout', 'responsive');
+    setStyles(image, {
+      'object-fit': 'cover',
+    });
+    const wrapper = this.element.ownerDocument.createElement('wrapper');
+    // This makes the non-iframe image appear in the exact same spot
+    // where it will be inside of the iframe.
+    setStyles(wrapper, {
```

Comment:
There still should be a top-level element (e.g. `div`) with `placeholder` attribute. It itself also children with arbitrary markup.